zerver/actions/realm_settings.py

```diff
@@ -486,3 +487,57 @@ def do_send_realm_reactivation_email(realm: Realm, *, acting_user: Optional[User
         language=language,
         context=context,
     )
+
+
+def do_send_realm_deactivation_email(realm: Realm, acting_user: Optional[UserProfile]) -> None:
+    # if realm is not deactivated, but changed the domain
+    if not acting_user:
+        return
+
+    event_time = timezone_now()
+    RealmAuditLog.objects.create(
+        realm=realm,
+        acting_user=acting_user,
+        event_type=RealmAuditLog.REALM_DEACTIVATION_EMAIL_SENT,
+        event_time=event_time,
+        extra_data=orjson.dumps(
+            {
+                RealmAuditLog.ROLE_COUNT: realm_user_count_by_role(realm),
+            }
```

Comment:
I suspect we don't need this `extra_data`; it's only intended for changes in the set of users that impact billing.

Actually we should just delete the audit log entry entirely -- this is just a side effect of deactivating the realm, and will happen automatically; at most we would add a note in the `extra_data` for the realm deactivation that the deactivation email was sent, or maybe as a string which variant was sent.